app/channels/event_channel.rb

```diff
@@ -47,6 +47,9 @@ def vote(data)
 
     # update only the voter with their vote value
     ActionCable.server.broadcast unique_channel, action: 'add-song', data: with_current_user_vote(song)
+
+    # remove song if score is less than -4
+    song.remove_from_queue if song.score < -4
```

Comment:
While song.remove_from_queue effectively removes the song, it also updates queued_at as if the song was actually played.

You may want to make another method in song that still broadcasts the remove like remove_from_queue does, but calls self.destroy instead of updating the queued_at time.